src/unitxt/operators.py

```diff
@@ -671,6 +672,69 @@ def process(self, stream: Stream, stream_name: str = None) -> Generator:
                 yield instance
 
 
+class FilterByCondsOnValues(SingleStreamOperator):
+    """
+    Filters a stream, yielding only instances that match conditions specified as stringed-lambda on values in the provided fields.
+
+    Args:
+        values (Dict[str, Any]): For each field, the values that instances should match to be included in the output.
+    """
+
+    required_values: Dict[str, Any] = {}
+
+    def process(self, stream: Stream, stream_name: str = None) -> Generator:
+        for instance in stream:
+            filter = False
+            for key, value in self.required_values.items():
+                filtlambda = eval(value)
+                if not key in instance:
+                    raise ValueError(
+                        f"Required filter field ('{key}') in FilterByCondsOnValues is not found in {instance}"
+                    )
+                if not filtlambda(instance[key]):
+                    filter = True
+            if not filter:
+                yield instance
+
+
+class FindAndStoreMostCommonValues(MultiStreamOperator):
+    field_name: str
+    stream_name: Optional[str] = "train"
+    top_percent: Optional[int] = 80
+    min_relative_freq_percent: Optional[int] = 5
+    new_field: Optional[str] = None
+    """
+    Given a MultiStream, a stream_name, and a field name, and two margins of commonality.
+    Do nothing (exception thrown) if the multistream does not contain a stream named stream_name, or if the multistream does not
+    contain a field (feature) named field_name.
+    Count the frequency of each value, over the stream named stream_name. Sort the values by frequency.
+    Discard each value that is not at the top_percent of that sorted list.
```

Comment:
Not sure if there is a reason to take first 80% of the list. One can always cut it later. Maybe I'm missing something.

Reply:
I accept. I now understand what you had in mind: measure only by frequency of coverage.